frontend/viewer/src/lib/entry-editor/field-editors/SingleOptionEditor.svelte

```diff
@@ -1,18 +1,85 @@
 <script lang="ts">
-  import CrdtOptionField from '../inputs/CrdtOptionField.svelte';
-  import FieldTitle from '../FieldTitle.svelte';
+  /* eslint-disable @typescript-eslint/no-duplicate-type-constituents, @typescript-eslint/no-redundant-type-constituents */
+  import { createEventDispatcher } from 'svelte';
+  import MapBind from '../../utils/MapBind.svelte';
+
+  import type { WritingSystemSelection } from '../../config-types';
+  import { useCurrentView } from '../../services/view-service';
   import { pickWritingSystems } from '../../utils';
-  import type {WritingSystemSelection} from '../../config-types';
-  import type { MenuOption } from 'svelte-ux';
-  import {useCurrentView} from '../../services/view-service';
-  import {useWritingSystems} from '../../writing-systems';
+  import { useWritingSystems } from '../../writing-systems';
+  import FieldTitle from '../FieldTitle.svelte';
+  import CrdtOptionField from '../inputs/CrdtOptionField.svelte';
+
+  type Value = $$Generic;
+  type Option = $$Generic;
+  type Id = Value extends undefined ? string | undefined : string;
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  type $$Props = {
+    id: string;
+    wsType: WritingSystemSelection;
+    name?: string;
+    readonly: boolean;
+    value: Value;
+    options: Option[];
+    getOptionLabel(option: Option): string;
+  }
+  & // mappings we support out of the box
+  (
+    { value: Id | { id: string }; } |
+    { getValueId: (value: Value) => Id; }
+  ) & (
+    { options: { id: string }[]; } |
+    { getOptionId: (value: Option) => string; }
+  ) & (
+    { value: Value & Option } |
+    { value: Id; idValue: true; } | // we need idValue or else we don't know what type to return at run time
+    { getValueById: (id: Id) => Value }
+  );
+
+  const dispatch = createEventDispatcher<{
+    change: { value: Value };
+  }>();
+
+  function onChange(): void {
+    // wait for the change to be mapped
+    setTimeout(() => dispatch('change', {value}));
+  }
 
   export let id: string;
   export let wsType: WritingSystemSelection;
   export let name: string | undefined = undefined;
   export let readonly: boolean = false;
-  export let value: string | undefined;
-  export let options: MenuOption<string>[];
+  export let value: Value;
+  export let options: Option[];
+
+  export let idValue: true | undefined = undefined;
```

Comment:
The problem is that there are 2 value<-->option mappings I'm trying to support out of the box. So, there needs to be something to tell the component which one should be used.

So, this field says: "please give me plain string IDs as values" instead of giving me the whole option itself as the value.
The compiler will force you to set this to true if you're binding a string value.

Reply:
how about `valueIsId`?

Reply:
There's also `valueIds` in the multi-option editor. It doesn't work quite as well when renamed to `valuesAreIds`, but...I do still like the clarity of it 👍.